### DISCUSSION TITLE
Feature: Chat GPT window

### DISCUSSION THREAD

**Discussion by Ruandv** [bluecaret/carettab#323] [Ideas and Feature Requests]
I would like to suggest adding a Chat GPT component to this wonderful extension.
I am using GPT on a daily basis and find it sometimes better to "chat" and ask for quick code snippets or interesting historic facts than to go and look through the internet.

This component would have to work similar to what the Loadshedding component does where the user would have to register on the site and then enter their API key.

**Comment by bluecaret** [comment]
This is something I definitely would like to add in! Thanks for the suggestion!

**Comment by Ruandv** [comment]
Would it be OK to follow the same approach as the loadshedding component for the API_KEY? 
Or do you have another way that should be used?

**Comment by bluecaret** [comment]
I appreciate the assistance, however I think I'd like to save this feature for after the rework I'm doing, I also want to give it some thought on how to show it given the many uses of AI.

For now keeping the work on the current version to a minimum. Thank you

**Comment by bluecaret** [comment]
This feature will be part of the upcoming update, though heads up, it will be a Premium Access feature.
